### PR TITLE
Port to JupyterLab 3

### LINKED ISSUE #32: jupyterlab 3.x compatibility
Hi

That would be great if you make extension compatible with jupyterlab 3.x

Thanks

### PR DESCRIPTION
**Pull request by fcollonval**
Fixes #32

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks!

Just rebased onto the latest master.

**Comment by fcollonval** [comment]
Thanks a lot Jeremy

File: setup.py [hunk 1]
```diff
@@ -1,86 +1,96 @@
...
+    skip_if_exists
 )
 import setuptools
 
-HERE = os.path.abspath(os.path.dirname(__file__))
+HERE = Path(__file__).parent.resolve()
 
 # The name of the project
-name="jupyterlab-snippets"
+name = "jupyterlab_snippets"
```
Comment (jtpio) [inline setup.py:19]: Should we keep `jupyterlab-snippets` here? (purely for esthetics reasons)

And maybe use:

```python
package = name.replace('-', '_')
```

For the folder and package names.

Comment (fcollonval) [reply setup.py]: Thanks for the suggestion. I made the modification

File: setup.py [hunk 2]
```diff
@@ -1,86 +1,97 @@
...
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "package.json").read_bytes())
 
 setup_args = dict(
-    name=name,
-    version=version,
-    url="https://github.com/QuantStack/jupyterlab-snippets",
-    author="QuantStack",
-    description="Code Snippets Extension for JupyterLab",
+    name=package,
```
Comment (jtpio) [inline setup.py:64]:
```suggestion
    name=name,
```

**Review by jtpio — APPROVED** [review]
Looks good, thanks!

**Pull request merged**